static/js/choropleth/choropleth.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import ReactDOM from "react-dom";
+import axios from "axios";
+import { MainPane } from "./choropleth_template";
+
+const d3 = require("d3");
+var url = new URL(window.location.href);
+
+/**
+ * Generates choropleth map from API on pageload.
+ */
+window.onload = () => {
+    const urlParams = new URLSearchParams(window.location.search);
+    ReactDOM.render(
+        React.createElement(MainPane),
+        document.getElementById("main-pane")
+    );
+
+    // Build api call from request.
+    var base_url = "choropleth/download?";
+    base_url += "statVar=" + urlParams.get("statVar");
+    base_url += "&perCapita=" + (urlParams.get("perCapita") == "true");
+    base_url += "&place=" + urlParams.get("place");
+
+    // Create request and generate map.
+    axios.get(base_url).then((resp) => {
```

Comment:
What happens if the get fails?

Reply:
The map doesn't load. We need to add some UI for failure.